pravega/src/main/scala/akka/stream/alpakka/pravega/impl/PravegaFlow.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.pravega.impl
+
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import io.pravega.client.ClientConfig
+import io.pravega.client.stream.{EventStreamWriter, EventWriterConfig, Serializer}
+
+import scala.util.control.NonFatal
+import akka.stream.ActorAttributes
+
+@InternalApi class PravegaFlowStageLogic[A](val shape: FlowShape[A, A],
+                                            val clientConfig: ClientConfig,
+                                            val scope: String,
+                                            topic: String,
+                                            writerConfig: EventWriterConfig,
+                                            serializer: Serializer[A])
+    extends GraphStageLogic(shape)
+    with PravegaCapabilities {
+
+  private def in = shape.in
+  private def out = shape.out
+
+  private var writer: EventStreamWriter[A] = _
+
+  /**
+   * Initialization logic
+   */
+  override def preStart(): Unit =
+    try {
+      writer = createWriter(topic, serializer)
+    } catch {
+      case NonFatal(ex) => failStage(ex)
+    }
+
+  setHandler(in, new InHandler {
+    override def onPush(): Unit = {
+      val msg = grab(in)
+
+      writer.writeEvent(msg)
```

Comment:
It would be nice to support the `writeEvent` overload that takes a `routingKey` so that we can provide ordering guarantees.  I understand there are lots of potential features that could be implemented and I don't want to suggest we implement much more than the basic functionality at this point, but I would imagine this is a common writing requirement.  WDYT?

Reply:
Definitively right.